File: EmployeePayslipCalculator/EmployeePayslipCalculator.Models/ResponseResult.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace EmployeePayslipCalculator.Models
{
    public class ResponseResult
    {
        /// <summary>
        /// Gets or sets a value indicating whether this instance is success.
        /// </summary>
        /// <value>
        ///   <c>true</c> if this instance is success; otherwise, <c>false</c>.
        /// </value>
        public bool IsSuccess { get; set; }
        /// <summary>
        /// Gets or sets the message.
        /// </summary>
        /// <value>
        /// The message.
        /// </value>
        public string Message { get; set; }

        public ResponseResult()
        {
            this.IsSuccess = true;
            Message = "OK";
        }

    }

    public class ResponseResult<T>: ResponseResult
    {
        /// <summary>
        /// Gets or sets the result.
        /// </summary>
        /// <value>
        /// The result.
        /// </value>
        public T Result { get; set; }
    }
}

File: EmployeePayslipCalculator/Presentation/EmployeePayslipCalculator.MobileApp/src/models/responseResult.ts
export class ResponseResultBase {
    /**
     * Gets or sets a value indicating whether this instance is success.
     * @value
     * @c  true if this instance is success; otherwise, @c  false.
     */
    isSuccess: boolean;
    /**
     * Gets or sets the message.
     * @value
     * The message.
     */
    message: string;
}

export class ResponseResult<T> extends ResponseResultBase {
    /**
     * Gets or sets the result.
     * @value
     * The result.
     */
    result: T;
}
